src/interfaces/python/factory_python.i

```diff
@@ -0,0 +1,184 @@
+%{
+#include <shogun/util/visitors/InterfaceTypeVisitor.h>
+
+// we can use:
+// SWIG_TypeQueryModule or SWIG_MangledTypeQueryModule or SWIG_Python_TypeQuery
+// to query the swig_type in run time and be less implementation dependent maybe?
+// but this is faster and should not cause problems
+#define SHOGUN_GET_SWIG_TYPE(name)                                             \
+	SWIGTYPE_p_std__shared_ptrT_shogun__##name##_t
```

Comment:
I am wondering if this can be generalised to the other swig interfaces?

Reply:
I'm assuming you mean the whole file. I would guess so. `factory_python` is a horrible name for that regard. Any name suggestions?

Reply:
yes, that's fine. And yes, I meant if you could for example template `ShogunInterfaceToPyObject`, and the template determines the interface object type, e.g. `ShogunInterfaceFactory<PyObject>`.

Reply:
I have two reasons not to:
1. Added complexity without any other use case at the moment.
2. After reading "GetterVisitor" I don't see any place for this visitor in the future (and maybe not for InterfaceClassVisitor either). Once the parameter framework and AnyVisitor support base types we can get rid of both new visitors, and use GetterVisitor directly, no?

Reply:
Since R is working now, I should make this generic as you suggested :"D

Reply:
I think so, that would be great! Then it could be easier to add other languages?

Reply:
Yeah. This should work with octave as well.